src/bin/ion/commands/generate/generator.rs

```diff
@@ -277,19 +278,55 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
         schema_system: &mut SchemaSystem,
     ) -> CodeGenResult<()> {
         for authority in authorities {
-            // Sort the directory paths to ensure nested type names are always ordered based
-            // on directory path. (nested type name uses a counter in its name to represent that type)
-            let mut paths = fs::read_dir(authority)?.collect::<Result<Vec<_>, _>>()?;
-            paths.sort_by_key(|dir| dir.path());
-            for schema_file in paths {
-                let schema_file_path = schema_file.path();
-                let schema_id = schema_file_path.file_name().unwrap().to_str().unwrap();
-
-                let schema = schema_system.load_isl_schema(schema_id).unwrap();
+            self.generate_code_for_directory(authority, None, schema_system)?;
+        }
+        Ok(())
+    }
 
-                self.generate(schema)?;
+    /// Helper method to generate code for all schema files in a directory
+    /// `relative_path` is used to provide a relative path to the authority for a nested directory
+    pub fn generate_code_for_directory<P: AsRef<Path>>(
+        &mut self,
+        directory: P,
+        relative_path: Option<&str>,
+        schema_system: &mut SchemaSystem,
+    ) -> CodeGenResult<()> {
+        // Sort the directory paths to ensure nested type names are always ordered based
+        // on directory path. (nested type name uses a counter in its name to represent that type)
```

Comment:
This comment is obsolete, right?

Reply:
Yes, we can remove this since now we dont depend on nested type name counter.